common/src/sql/dbinit.sql

```diff
@@ -152,8 +148,9 @@ CREATE TABLE omicron.public.sled_resource (
 );
 
 -- Allow looking up all resources which reside on a sled
-CREATE INDEX ON omicron.public.sled_resource (
-    sled_id
+CREATE UNIQUE INDEX ON omicron.public.sled_resource (
```

Comment:
This is one of the cases mentioned in the PR description.  If we want tp aginate the sled_resources on a sled, we need the "id" here, too.